src/components/Link.tsx

```diff
@@ -175,8 +181,14 @@ export function useLink({
     }
   }, [disableMismatchWarning, displayText, href, isExternal, openModal])
 
-  const onLongPress =
-    isNative && isExternal && shareOnLongPress ? handleLongPress : undefined
+  const onLongPress = React.useCallback(
+    (e: GestureResponderEvent) => {
+      const exitEarlyIfFalse = outerOnLongPress?.(e)
+      if (exitEarlyIfFalse === false) return
+      return isNative && shareOnLongPress ? handleLongPress : undefined
+    },
+    [outerOnLongPress, handleLongPress, shareOnLongPress],
+  )
```

Comment:
No real change here, but of course you can exit early now.

Dropped the `isExternal` because it's only intended to be used to show the old `link-warning` modal. `shareOnLongPress` is what should determine if the `handleLongPress` handler is actually passed into the component.